File: solutions/java/0206-reverse-linked-list.java
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode prev = null;
        while (head != null) {
            ListNode next_node = head.next;
            head.next = prev;
            prev = head;
            head = next_node;
        }
        return prev;
    }
}

File: solutions/cpp/0206-reverse-linked-list.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* prev = NULL;
        while (head) {
            ListNode* next_node = head->next;
            head->next = prev;
            prev = head;
            head = next_node;
        }
        return prev;
    }
};
